src/communication/xcConnection.ts

```diff
@@ -34,7 +29,9 @@ export class DefaultConnection implements Connection {
             log.debug("getModel request failed");
             log.debug(err);
         };
-        session.init(openListener, errorListener);
+        let closeListener = (_: CloseEvent) => {
```

Comment:
First, I'm not sure to know how this feature should work.
- Reconnection should be based on a circuit breaker pattern. (Why keep on trying to reconnect intensively if the server is down for a long period of time ?)
- I think the user should know what's happening. I explain. If the connection goes down and you reconnect, I could miss important updates. So perhaps I want to to a snapshot (depends on use cases) when the connection is up again